lib/liquid/reversable_range.rb

```diff
@@ -0,0 +1,77 @@
+module Liquid
+  class ReversableRange
+    include Enumerable
+
+    def initialize(min, max)
+      @min = min
+      @max = max
+      @reversed = false
+    end
+
+    def each
+      if reversed
+        index = max
```

Comment:
It's likely a premature optimization, but using ivars directly is a fair bit faster in general.

```
Warming up --------------------------------------
         attr_reader   147.241k i/s -    150.359k times in 1.021174s (6.79μs/i)
                ivar   247.108k i/s -    249.557k times in 1.009912s (4.05μs/i)
Calculating -------------------------------------
         attr_reader   143.032k i/s -    441.723k times in 3.088285s (6.99μs/i)
                ivar   255.390k i/s -    741.323k times in 2.902711s (3.92μs/i)

Comparison:
                ivar:    255389.9 i/s
         attr_reader:    143031.8 i/s - 1.79x  slower
```

with

```rb
    def each
      if reversed
        index = @max
        while index >= @min
          yield index
          index -= 1
        end
      else
        index = @min
        while index <= @max
          yield index
          index += 1
        end
      end
    end
```

using `ReversableRange.new(1, 100)`